docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -1,466 +1,1741 @@
+
+-  Application developers: to query the ledger to display information to
+   users, integrate or carry out automated business processes, and
+   debugging.
+-  Business analysts: to access and analyze ledger data with ad-hoc
+   investigations to understand operational usage.
+-  Support analysts: to diagnose any problems on a live system, when
+   sophisticated inspection of the ledger is required.
+-  Application operators: to take a snapshot of the ledger, either with
+   or without history in order to analyse and investigate.
+-  Report analysts: to provide access to ledger data information for
+   analysis and aggregation with standard reporting tools, on a
+   historical or real-time basis.
+
+Purpose
+~~~~~~~
+
+PQS can be described as a kind of
+`bridge <https://en.wikipedia.org/wiki/Bridge_pattern>`__ connecting the
+ledger with a datastore. Providing read access to the data via a
+traditional SQL database decouples the reader from a specific
+Participant and enables users to leverage their existing tools and
+knowledge.
+
+The Participant's primary role is to submit and propogate transactions
+in the network, which requires interaction to submit commands. PQS
+sources the transaction ledger from the Participant on a streaming
+basis, maintains history and state based on the events, and makes the
+data available for users to query. Therefore, it alleviates the need for
+the Participant to be involved in read-loads and can be scaled
+independently.
+
+Operational Architecture
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+PQS is a standalone application that is configured via command line
+arguments, environment variables and configuration files. It is intended
+to run as a long-running process and is resilient to network outages. It
+is also designed to be idempotent (or "crash tollerant") so that it can
+be restarted safely at any time. It does not write anything to the
+ledger but is only a passive consumer of data from a Canton Participant.
+
+The following diagram shows that PQS initiates connections to both the
+participant and the datastore: (arrows indicate the direction of
+connection)
+
+|diagram|
+
+Similarly, from the perspective of data-flow:
+
+|image1|
+
+Expanding the application node from the above, you can imagine many
+potential architectures, including:
+
+|image2|
+
+Key Feature Summary
+~~~~~~~~~~~~~~~~~~~
+
+PQS enables the following key capabilities:
+
+-  Connect to the ledger and replicate a snapshot of the contract state
+   at a given ledger offset, which can be nominated.
+-  Continue to update the datastore with subsequent events up to a
+   nominated offset or indefinitely.
+-  Filter the scope by Party and/or Daml template.
+-  Embed an API within the database to allow consumers to easily access
+   the ledger data by:
+
+   -  Using familiar Daml identifiers (eg.
+      ``mypackage:My.Module:MyTemplate``) instead of PostgreSQL
+      approximations.
+   -  Preserving Party based privacy constraints.
```

Comment:
Could you explain what this means?  I suspect that you mean on the ingest path.  However a reader can assume it does this on the egress (SQL) path.

Reply:
The intent was that you can easily preserve the 'show me what is visible to party X' - but we don't do enough to make this claim yet so I'll remove.